Detectors/TPC/reconstruction/src/HwClusterer.cxx

```diff
@@ -215,24 +215,24 @@ void HwClusterer::process(std::vector<o2::tpc::Digit> const& digits, MCLabelCont
     index = mapTimeInRange(digit.getTimeStamp()) * mPadsPerRowSet[mGlobalRowToRowSet[digit.getRow()]] + (digit.getPad() + 2);
     // offset of digit pad because of 2 empty pads on both sides
 
+    float charge = digit.getChargeFloat();
+
     // TODO: fill noise here as well if necessary
     if (mPedestalObject) {
-      /*
-       * If a pedestal object was registered, check if charge of pad is greater
-       * than pedestal value. If so, assign difference of charge and pedestal
-       * to buffer, if not, set buffer to 0.
-       */
-      if (digit.getChargeFloat() < mPedestalObject->getValue(CRU(digit.getCRU()), digit.getRow(), digit.getPad())) {
-        mDataBuffer[mGlobalRowToRowSet[digit.getRow()]][index][mGlobalRowToVcIndex[digit.getRow()]] = 0;
-      } else {
-        mDataBuffer[mGlobalRowToRowSet[digit.getRow()]][index][mGlobalRowToVcIndex[digit.getRow()]] = static_cast<unsigned>(
-          (digit.getChargeFloat() - mPedestalObject->getValue(CRU(digit.getCRU()), digit.getRow(), digit.getPad())) * (1 << 4));
-      }
-    } else {
-      mDataBuffer[mGlobalRowToRowSet[digit.getRow()]][index][mGlobalRowToVcIndex[digit.getRow()]] = static_cast<unsigned>(digit.getChargeFloat() * (1 << 4));
+      charge -= mPedestalObject->getValue(CRU(digit.getCRU()), digit.getRow(), digit.getPad());
     }
-    if (mDataBuffer[mGlobalRowToRowSet[digit.getRow()]][index][mGlobalRowToVcIndex[digit.getRow()]] > 0x3FFF)
+    /*
+     * charge could be smaller than 0 due to pedestal subtraction, if so set it to zero
+     * noise thresholds for zero suppression could also be done here ...
+     */
+    if (charge < 0) {
+      charge = 0;
+    }
+
+    mDataBuffer[mGlobalRowToRowSet[digit.getRow()]][index][mGlobalRowToVcIndex[digit.getRow()]] = static_cast<unsigned>(charge * (1 << 4));
+    if (mDataBuffer[mGlobalRowToRowSet[digit.getRow()]][index][mGlobalRowToVcIndex[digit.getRow()]] > 0x3FFF) {
       mDataBuffer[mGlobalRowToRowSet[digit.getRow()]][index][mGlobalRowToVcIndex[digit.getRow()]] = 0x3FFF; // set only 14 LSBs
+    }
```

Comment:
I think to avoid unnecessary multiplication for sub-threshold charges and unnecessary comparisons, it is more efficient to do
````
  mDataBuffer[mGlobalRowToRowSet[digit.getRow()]][index][mGlobalRowToVcIndex[digit.getRow()]] =
    charge < 0 ? 0 : ( (charge < float(0x3FFF)/(1<<4)) ? (charge*(1<<4)) : 0x3FFF );
````

Reply:
why `charge < float(0x3FFF)/(1<<4)` and not `charge * (1<<4)< float(0x3FFF)`?

Reply:
Ahh I guess because the compiler can precalculate `float(0x3FFF)/(1<<4)` ok.

Reply:
yes, the constant will be calculated at compilation time.